core/packages/contracts/src/utils/Bitfield.sol

```diff
@@ -5,10 +5,13 @@ pragma solidity 0.8.20;
 import "./Bits.sol";
 
 library Bitfield {
+    using Bits for uint256;
+
     /**
      * @dev Constants used to efficiently calculate the hamming weight of a bitfield. See
      * https://en.wikipedia.org/wiki/Hamming_weight#Efficient_implementation for an explanation of those constants.
      */
+
```

Comment:
Agree it's not a security risk, also notice you've added these constraints in comments.  
Btw:  based on the algorithm in `minimumSignatureThreshold` n is always no more than 2/3 of validators  and `prior` in `submitFinal*` should be same as what original from `createInitialBitfield` which is also checked more than 2/3 of validators.  So `n <= countSetBits(prior)` should always be true if relayer follow the right way to construct calldata.  Then we may not need to repeatly add these constraints multiple times.

Reply:
Ah, I see: in `verifyCommitment` (also internal), we have `signatureCount < 2/3*vset.length` (because of `minimumSignatureThreshold`) and `2/3*vset.length < countSetBits(bitfield)` (because of `submitInitial*` and checking the bitfield hash against the ticket). Which means `n < countSetBits(prior)` so you're right here, we don't need to surface this condition to `submitFinal*`.

The only other place `subsample` is called is `createFinalBitfield` - might be worth surfacing the conditions on `bitfield` there, but that's minor.

Reply:
For `createFinalBitfield` seems we're not checking bitfield in calldata is the same initial bitfield, will that be a problem? 

so I just added another condition here https://github.com/Snowfork/snowbridge/pull/877/commits/b4d06bb647ecb0ee13539a770cd7df22b325a61f

Then it will just resolve the two concerns you mentioned above.